dvc/stage/__init__.py

```diff
@@ -121,7 +121,7 @@ def __init__(
 
         self.repo = repo
         self._path = path
-        self.cmd = cmd
+        self.cmd = " && ".join(cmd) if isinstance(cmd, list) else cmd
```

Comment:
Will && work the same in all shells?

Also, this is will make the `repro` create a long and ugly `cmd1 && cmd2 && cmd3 ...` in the dvc file instead of previously present nice list of commands. Doesn't seem like a good behavior.

Reply:
The behaviour should be similar to the multiline string, rather than this `&&`.

This approach will fail on serialization/deserialization, so we need to support reading this command as a list directly on the following:
https://github.com/iterative/dvc/blob/6237a665c355020bf9c083851c8a89e874d2d437/dvc/stage/run.py#L47

And, also support list as `cmd` on the `dvc.lock` file.

Reply:
cmd2,3,4... would stop if cmd1 returns some number else than 0.

Reply:
> cmd2,3,4... would stop if cmd1 returns some number else than 0.

I think it is the desired behavior as stated in the [issue](https://github.com/iterative/dvc/issues/4373): 

> This looks great and I think it should behave like cmd1 && cmd2 && cmd3, meaning that if one command fails then the rest is not executed and results in a stage failure.